### ISSUE TITLE
Missing dependency

### ISSUE THREAD

**Issue by rauchg** [alexanderGugel/ied#28]
First of all: wow, `ied` is remarkably fast. I love the nix-like architecture, and the small size of the codebase.

I was trying it out with `socket.io` and ran into an issue with a missing `engine.io` dependency. It also seems to not be building binary deps? (I actually think opting into these, and leaving them out by default would be swell!)

Here's a gifcast showing my problem
![ied](https://cloud.githubusercontent.com/assets/13041/11255754/cea7ee4c-8dfc-11e5-903d-93f2b6d828fa.gif)

**Comment by rauchg** [comment]
Also, perhaps crucially, the screencast shows the progress bar reaching 100 and then regressing to 97.xx. Race condition?

**Comment by alexanderGugel** [comment]
Hello @rauchg!

The issue is that ied currently only supports plain, old installs of npm packages. It looks like `engine.io` is [declared](https://github.com/socketio/socket.io/blob/master/package.json#L24) as a GitHub dependency. This is currently not supported.

I'm working on supporting all kinds of packages as we speak. This change will include lifecycle events (such as postinstall). I expect to have that done by the end of this week.

You can see the current progress on that side here: https://github.com/alexanderGugel/ied/issues/7#issuecomment-157316751

Concerning the progress bar: Determining progress on installations is hard to get right. It's not really a race condition, but rather that the package has been resolved (as "in the `package.json` has been fetched from the registry", but the download hasn't kicked in yet).

Truth being told, the progress bar sucks currently. It took npm four years to get that part right, now I understand why :smile: 

**TL;DR** - We're working on it and any help is appreciated!

P.S.: It's an honour to have you here!

**Comment by rauchg** [comment]
The published version of socket.io I believe points to a fixed engine.io
version.

Thank you!!

**Comment by rauchg** [comment]
Indeed

```
 ▲  ~ npm show socket.io dependencies

{ 'engine.io': '1.5.4',
  'socket.io-parser': '2.2.4',
  'socket.io-client': '1.3.7',
  'socket.io-adapter': '0.3.1',
  'has-binary-data': '0.1.3',
  debug: '2.1.0' }
```

**Comment by iamstarkov** [comment]
apparently, its still an issue (at least for me and [generator-travis](https://github.com/iamstarkov/generator-travis)):

```
➜  generator-travis git:(master) less package.json
  …
  "dependencies": {
    "merge-and-concat": "^1.0.0",
    "ramda": "^0.18.0",
    "sort-object": "^3.0.0",
    "travis-config-keys": "^1.0.0",
    "yamljs": "^0.2.4",
    "yeoman-generator": "^0.20.0"
  },
  "devDependencies": {
    "mocha": "*"
  }
➜  generator-travis git:(master) rm -rf node_modules
➜  generator-travis git:(master) ied i
➜  generator-travis git:(master) ied test
module.js:329
    throw err;
    ^

Error: Cannot find module 'ejs'
➜  generator-travis git:(master) rm -rf node_modules
➜  generator-travis git:(master) npm i
➜  generator-travis git:(master) npm test

> generator-travis@1.1.6 test /Users/vlasta/projects/generator-travis
> mocha



  travis:app
    ✓ creates config
    ✓ extends existing config
    ✓ uses config from options
    ✓ uses config with extra node versions from options


  4 passing (112ms)
```

**Comment by alexanderGugel** [comment]
Yes, it is still an issue. I didn't fix it, yet and can't guarantee that I will get to it before January.

**Comment by iamstarkov** [comment]
its okay, its still faster than npm. can you give some direction to solve this issue or maybe write fail-test, which everybody can work with?

**Comment by iamstarkov** [comment]
@alexanderGugel any suggestions?

**Comment by just-boris** [comment]
@iamstarkov have you tried the latest version `0.4.3` yet? It has a lot of changes, at least more descriptive error messages sometimes.

**Comment by mgcrea** [comment]
Can't reproduce against master:

```
$ ied i -b socket.io
$ ll node_modules/
total 4.0K
lrwxr-xr-x 1 olivier staff 53 Jul  1 17:32 socket.io -> .cas/e576f330cd0bed64e55b3fd26df991141884867b/package/
$ node
> require('socket.io')
{ [Function: Server] listen: [Circular] }
> 
```
